src/RoslynInsertionTool/RoslynInsertionTool/CoreXT.cs

```diff
@@ -39,27 +40,35 @@ internal class CoreXT
         private const string PackagePropsPath = "Directory.Packages.props";
         private const string LegacyPackagePropsPath = "Packages.props";
 
-        private readonly string _defaultConfigOriginal;
+        private readonly string? _defaultConfigOriginal;
 
-        public XDocument ConfigDocument { get; }
+        public XDocument? ConfigDocument { get; }
 
-        private CoreXT(string configOriginalText)
+        private CoreXT(string? configOriginalText)
         {
             _defaultConfigOriginal = configOriginalText;
-            ConfigDocument = XDocument.Parse(configOriginalText, LoadOptions.None);
+            ConfigDocument = configOriginalText is null ? null : XDocument.Parse(configOriginalText, LoadOptions.None);
         }
 
         public static async Task<CoreXT> Load(GitHttpClient gitClient, string commitId)
         {
             var vsRepoId = RoslynInsertionTool.VSRepoId;
             var vsBranch = new GitVersionDescriptor { VersionType = GitVersionType.Commit, Version = commitId };
 
-            using var defaultConfigStream = await gitClient.GetItemContentAsync(
-                vsRepoId,
-                DefaultConfigPath,
-                download: true,
-                versionDescriptor: vsBranch);
-            var defaultConfigOriginal = await new StreamReader(defaultConfigStream).ReadToEndAsync();
+            string? defaultConfigOriginal;
+            try
+            {
+                using var defaultConfigStream = await gitClient.GetItemContentAsync(
+                    vsRepoId,
+                    DefaultConfigPath,
+                    download: true,
+                    versionDescriptor: vsBranch);
+                defaultConfigOriginal = await new StreamReader(defaultConfigStream).ReadToEndAsync();
+            }
+            catch (VssServiceException ex) when (ex.IsFileNotFound())
+            {
+                defaultConfigOriginal = null;
```

Comment:
consider logging something when there is no default.config (I assume this is expected for main, but unexpected for servicing brnaches)?

Reply:
Done